gatsby-node.ts

```diff
@@ -0,0 +1,177 @@
+import { GatsbyNode } from "gatsby"
+import path from "path"
+import { createFilePath } from "gatsby-source-filesystem"
+import { parse } from "svgson"
+import { pathThatSvg } from "path-that-svg"
+import fs from "fs"
+
+export const createPages: GatsbyNode["createPages"] = async ({
+  actions,
+  graphql,
+  reporter,
+}) => {
+  const { createPage } = actions
+
+  // Filter only for md files in the src/pages directory. Any md files found
+  // outside of that should not have a page created for it
+  const result = await graphql(`
+    {
+      allMarkdownRemark(filter: { fileAbsolutePath: { regex: "/(pages)/" } }) {
+        edges {
+          node {
+            id
+            fields {
+              slug
+            }
+            frontmatter {
+              template
+            }
+          }
+        }
+      }
+    }
+  `)
+
+  // Handle errors
+  if (result.errors) {
+    reporter.panicOnBuild(`Error while running GraphQL query.`)
+    return
+  }
+
+  // @ts-ignore
+  result.data.allMarkdownRemark.edges.forEach(({ node }) => {
+    const templateName = node.frontmatter.template || `default`
+    const template = path.resolve(
+      path.join("src/templates/", `${templateName}.tsx`)
+    )
+    createPage({
+      path: node.fields.slug,
+      component: template,
+      context: { id: node.id }, // additional data can be passed via context
+    })
+  })
+}
+
+export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] =
+  ({ actions }) => {
+    const { createTypes } = actions
+    const typeDefs = `
+    type MarkdownRemark implements Node {
+      frontmatter: Frontmatter
+    }
+    type Frontmatter {
+      hero: Hero
+    }
+    type Hero {
```

Comment:
Do we need to register all of our info in this definition?  For example,  see Hero, CTA, etc. Why don't we need to register the NavBar info? or any other new pages/sections we add?

Reply:
I believe it's unnecessary because external plugins will add types under the hood based on the `static/admin/config.yaml` or markdown files(I'm not sure here 🤔 ). We just need to keep the `SVGAttributes` and `SVG` because it's our custom implementation.